pandas/core/dtypes/dtypes.py

```diff
@@ -144,22 +153,22 @@ def __bytes__(self):
         encoding = get_option("display.encoding")
         return self.__unicode__().encode(encoding, 'replace')
 
-    def __repr__(self):
+    def __repr__(self) -> str_type:
         """
         Return a string representation for a particular object.
         """
         return str(self)
 
-    def __hash__(self):
+    def __hash__(self) -> int:
         raise NotImplementedError("sub-classes should implement an __hash__ "
                                   "method")
 
-    def __getstate__(self):
+    def __getstate__(self) -> Dict[str_type, Any]:
```

Comment:
Would like to minimize the use of `Any` as much as possible. Can't the value here not be `str` as well? May need to add type to `self._metadata` to really make use of this

Reply:
I think `Dict[str_type, Union[Optional[Index], bool]]` will do it.

Reply:
``__getstate__`` is also used by subclasses, so the strings in _metadata that I find are needed are (with associated guess for types for attributes):
* "categories": Optional[Index]
* "ordered": Optional[bool]
* "unit": str (+other?)
* "tz": str, I think (+ others?)
* "freq": Not sure ATM
* "subtype": Not sure ATM (``_typing.Dtype``?)

If the type hints are important to be specific to the ExtensionsDtype, maybe subclass ``__getstate__`` in order to set the correct type hints? (in different PR IMO)?